internal/pkg/daemon/bpfrecorder/bpf/recorder.bpf.c

```diff
@@ -112,11 +113,9 @@ static __always_inline int save_args(u32 event_id, const char * filename,
     return 0;
 }
 
-static __always_inline int load_args(u32 event_id, const char ** filename,
-                                     u64 * flags)
+static __always_inline int load_args(const char ** filename, u64 * flags)
 {
-    u32 tgid = bpf_get_current_pid_tgid();
-    u64 id = ((u64)event_id << 32) | tgid;
+    u32 id = bpf_get_current_pid_tgid();
```

Comment:
Is this state map just used internally in eBPF to keep track of arguments when the syscall enters and exits the trace point? Is the event id not anymore needed? I suppose was not parsed in the Go userspace code since I don't see any change in there.

Reply:
> Is this state map just used internally in eBPF to keep track of arguments when the syscall enters and exits the trace point?

Correct. This is just temporary storage so that we have the filename in the exit hook.

> Is the event id not anymore needed?

From chatting with @0xmilkmix (who wrote the first iteration): Initially this code used the PID as storage key only, but that was causing race conditions with concurrent syscalls from different threads. A first attempt to fix this was to use PID + event id (=syscall name) as a the key. Only later on @0xmilkmix discovered pid_tgid, which uniquely identifies threads. With the proper fix in place, the event id is not needed anymore. :)